internal/attestation/crafter/materials/csaf.go

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package materials
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	api "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
+	"github.com/chainloop-dev/chainloop/internal/schemavalidators"
+	"github.com/rs/zerolog"
+)
+
+// CSAFCrafter is a crafter for CSAF VEX, CSAF Informational Advisory, CSAF Security Incident Response, and CSAF Security Advisory
+// material types. It implements the Crafter interface.
+type CSAFCrafter struct {
+	backend *casclient.CASBackend
+	*crafterCommon
+}
+
+// NewCSAFVEXCrafter creates a new CSAF VEX crafter
+func NewCSAFVEXCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_VEX {
+		return nil, fmt.Errorf("material type is not CSAF_VEX format")
+	}
+
+	return baseCSAFCrafter(materialSchema, backend, l)
+}
+
+// NewCSAFInformationalAdvisoryCrafter creates a new CSAF Informational Advisory crafter
+func NewCSAFInformationalAdvisoryCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_INFORMATIONAL_ADVISORY {
+		return nil, fmt.Errorf("material type is not CSAF_INFORMATIONAL_ADVISORY format")
+	}
+
+	return baseCSAFCrafter(materialSchema, backend, l)
+}
+
+// NewCSAFSecurityIncidentResponseCrafter creates a new CSAF Security Incident Response crafter
+func NewCSAFSecurityIncidentResponseCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
```

Comment:
mm, I am not sure I understand. Will this fail if I provide a VEX file but I say that's a INCENT_REPORT?